### PR TITLE
Implemented swizzle for neon and neon64

### PR DESCRIPTION
**Pull request by JohanMabille**
~~This PR is not ready yet, I will rebase it when #649 is merged so it can leverage the tests.
This is mainly for review and discussion.~~

Rebase done.

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@JohanMabille for the formatting part, you need to run

```
find -name '*.[ch]pp' -exec clang-format -i {} \;
```

from the top source dir.

**Comment by JohanMabille** [comment]
@serge-sans-paille I think this is ready to be merged

**Comment by JohanMabille** [comment]
Now it is ;)

File: include/xsimd/arch/xsimd_neon.hpp [hunk 1]
```diff
@@ -2296,6 +2296,112 @@ namespace xsimd
...
+                                          requires_arch<neon>) noexcept
+        {
+            std::array<uint32_t, 4> data;
+            store_aligned(data.data(), self, A());
+            return set(batch<uint32_t, A>(), A(), data[V0], data[V1], data[V2], data[V3]);
+        }
+
+        template <class A, uint32_t V0, uint32_t V1, uint32_t V2, uint32_t V3>
+        inline batch<int32_t, A> swizzle(batch<int32_t, A> const& self,
+                                         batch_constant<batch<uint32_t, A>, V0, V1, V2, V3>,
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_neon.hpp:2367]: These kernels are generic, there should be a single variadic version in generic, or am I missing something?

File: include/xsimd/arch/xsimd_neon.hpp [hunk 2]
```diff
@@ -2296,6 +2296,112 @@ namespace xsimd
...
+
+        template <class A, uint8_t V0, uint8_t V1, uint8_t V2, uint8_t V3, uint8_t V4, uint8_t V5, uint8_t V6, uint8_t V7,
+                  uint8_t V8, uint8_t V9, uint8_t V10, uint8_t V11, uint8_t V12, uint8_t V13, uint8_t V14, uint8_t V15>
+        inline batch<uint8_t, A> swizzle(batch<uint8_t, A> const& self,
+                                         batch_constant<batch<uint8_t, A>, V0, V1, V2, V3, V4, V5, V6, V7,
+                                                        V8, V9, V10, V11, V12, V13, V14, V15>,
+                                         requires_arch<neon>) noexcept
+        {
+            std::array<uint8_t, 16> data;
+            store_aligned(data.data(), self, A());
```
[the remark this answers is not in the document]
Comment (JohanMabille) [reply include/xsimd/arch/xsimd_neon.hpp:2317]: It does not matter for neon, aligned and unaligned store call the same instruction in the end.

File: include/xsimd/arch/xsimd_neon.hpp (hunk 1 above)
Comment (JohanMabille) [reply include/xsimd/arch/xsimd_neon.hpp:2367]: Indeed

**Comment by JohanMabille** [comment]
@serge-sans-paille this is now ready.

File: include/xsimd/arch/xsimd_neon.hpp [hunk 3]
```diff
@@ -2296,6 +2296,34 @@ namespace xsimd
...
+
+        template <class A, class T, class I, I... idx>
+        inline batch<T, A> swizzle(batch<T, A> const& self,
+                                   batch_constant<batch<I, A>, idx...>,
+                                   requires_arch<neon>) noexcept
+        {
+            std::array<T, batch<T, A>::size> data;
+            store_aligned(data.data(), self, A());
+            return set(batch<T, A>(), A(), data[idx]...);
+        }
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_neon.hpp:2317]: So much more beautiful

**Comment by serge-sans-paille** [comment]
Thanks for going through this :-)

**Pull request merged**